llvm/lib/CodeGen/GlobalISel/GISelKnownBits.cpp

```diff
@@ -565,7 +565,9 @@ void GISelKnownBits::computeKnownBitsImpl(Register R, KnownBits &Known,
     KnownBits ExtKnown = KnownBits::makeConstant(APInt(BitWidth, BitWidth));
     KnownBits ShiftKnown = KnownBits::computeForAddSub(
         /*Add=*/false, /*NSW=*/false, /* NUW=*/false, ExtKnown, WidthKnown);
-    Known = KnownBits::ashr(KnownBits::shl(Known, ShiftKnown), ShiftKnown);
+    Known = KnownBits::ashr(KnownBits::shl(Known, ShiftKnown), ShiftKnown,
+                            /*ShAmtNonZero=*/false,
+                            /*Exact*/ true);
```

Comment:
Could you read the `Exact` flag from the llvm::MachineInstr? Same for NSW and NUW above. Maybe I am missing something.

Reply:
I am extremely sceptical that this change will make any improvement to the KnownBits analysis for G_SBFX. I'd prefer to just remove this part from the patch.

Reply:
> Could you read the Exact flag from the llvm::MachineInstr?

No, this is not related to whether the G_SBFX itself is exact. This is only exact because we're operating on the result of extractBits, which will leave zero bits outside of the masked value.